Feature/linking context

Taiiiiga opened the pull request.

Taiiiiga committed:
LinkingContext

Finish the linking context

ttgc committed:
Merge branch 'dev' into feature/LinkingContext

Taiiiiga committed:
Private variable

ttgc commented on main/include/LinkingContext.hpp:
```diff
@@ -0,0 +1,26 @@
+	void deletePointer(GameObject* obj);
+	void addGameObject(GameObject* obj);
+
+	std::optional<int> getIDFromObject(GameObject* obj);
```
```suggestion
	std::optional<NetworkID> getIDFromObject(GameObject* obj);
```

ttgc commented on main/include/LinkingContext.hpp:
```diff
@@ -0,0 +1,26 @@
+{
+public:
+
+	LinkingContext(std::map<NetworkID, GameObject*> idObjt, std::map<GameObject*, ReplicationClassID> pointObjt);
```
```suggestion
	LinkingContext();
```

ttgc commented on main/src/LinkingContext.cpp:
```diff
@@ -0,0 +1,54 @@
+using NetworkID = uint32_t;
+
+
+LinkingContext::LinkingContext(std::map<NetworkID, GameObject*> idObjt, std::map<GameObject*, ReplicationClassID> pointObjt){
```
```suggestion
LinkingContext::LinkingContext : currentId(0), idObjt(), pointObjt()
{
```

ttgc commented on main/include/LinkingContext.hpp:
```diff
@@ -0,0 +1,26 @@
+
+	LinkingContext(std::map<NetworkID, GameObject*> idObjt, std::map<GameObject*, ReplicationClassID> pointObjt);
+
+	void addGameObjectAndID(NetworkID id, GameObject obj);
```
```suggestion
	void addGameObjectAndID(NetworkID id, GameObject* obj);
```

ttgc commented on main/src/LinkingContext.cpp:
```diff
@@ -0,0 +1,54 @@
+
+}
+
+void LinkingContext::addGameObjectAndID(NetworkID id, GameObject obj)
```
```suggestion
void LinkingContext::addGameObjectAndID(NetworkID id, GameObject* obj)
```

ttgc commented on main/src/LinkingContext.cpp:
```diff
@@ -0,0 +1,54 @@
+{
+	
+	//std::pair<int, GameObject*>(1, &obj);
+
```
Useless piece of code, need to be removed

ttgc commented on main/src/LinkingContext.cpp:
```diff
@@ -0,0 +1,54 @@
+	
+	//std::pair<int, GameObject*>(1, &obj);
+
+	idObjt.insert(std::pair<NetworkID, GameObject*>(id, &obj));
```
```suggestion
	idObjt.insert(std::pair<NetworkID, GameObject*>(id, obj));
```

ttgc commented on main/src/LinkingContext.cpp:
```diff
@@ -0,0 +1,54 @@
+	//std::pair<int, GameObject*>(1, &obj);
+
+	idObjt.insert(std::pair<NetworkID, GameObject*>(id, &obj));
+	pointObjt.insert(std::pair<GameObject*, NetworkID>(&obj, id));
```
```suggestion
	pointObjt.insert(std::pair<GameObject*, NetworkID>(obj, id));
```

ttgc commented on main/src/LinkingContext.cpp:
```diff
@@ -0,0 +1,54 @@
+{
+	std::map<GameObject*, NetworkID>::iterator deleted;
+
+	deleted = pointObjt.find(obj);
```
```suggestion
	const auto deleted = pointObjt.find(obj);
```

ttgc commented on main/src/LinkingContext.cpp:
```diff
@@ -0,0 +1,54 @@
+	idObjt.insert(std::pair<NetworkID, GameObject*>(currentId, obj));
+	pointObjt.insert(std::pair<GameObject*, NetworkID>(obj, currentId));
+
+	currentId++;
```
Managing new ID should be changed

ttgc commented on main/src/LinkingContext.cpp:
```diff
@@ -0,0 +1,54 @@
+	deleted = pointObjt.find(obj);
+
+	if (deleted != pointObjt.end())
+		pointObjt.erase(deleted);
```
Need also to be erased from the other map

Taiiiiga committed:
Correcting errors in Linking Context

Adding manager for new IDs, and for delete of gameobject (deleting from the other map)

Taiiiiga committed:
Linking Context correction

Taiiiiga committed:
Hoping everything is good

Taiiiiga committed:
Maybe this is the right time

ttgc reviewed (approved):
Let's merge !
